File: spring-boot-2-rest/src/main/java/com/example/springboot2rest/repository/CityRepository.java
package com.example.springboot2rest.repository;

import java.util.Optional;

import com.example.springboot2rest.model.City;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface CityRepository extends JpaRepository<City, Integer> {

    Page<City> findByStateId(Integer stateId, Pageable pageable);

    Page<City> findByStateUf(String uf, Pageable pageable);

    Page<City> findByStateRegionId(Integer regionId, Pageable pageable);

    Page<City> findByNameContaining(String name, Pageable pageable); // like '%name%'

    Optional<City> findByCod(Integer cod);
}

File: spring-boot-webmvc-redis-cache/src/main/kotlin/com/github/andrelugomes/repository/CityRepository.kt
package com.github.andrelugomes.repository

import com.github.andrelugomes.model.City
import com.github.andrelugomes.model.CitySerializable
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.repository.PagingAndSortingRepository
import org.springframework.stereotype.Repository

@Repository
interface CityRepository : PagingAndSortingRepository<City, String> {

    fun findAllByUf(uf: Int, pageable: Pageable): Page<City>
}